database/service_db.go

```diff
@@ -64,52 +87,119 @@ func (d *LevelDBServiceDB) All() ([]*service.Service, error) {
 				logrus.WithField("service", decodeErr.ID).Warning(decodeErr.Error())
 				continue
 			}
+			iter.Release()
 			return nil, err
 		}
 		services = append(services, s)
 	}
 	iter.Release()
```

Comment:
it's better to call this with a differ before starting to iterate

Reply:
i'm not sure, because then we call `iter.Error()`. so release may populate some error.